registry/server/appRoutes/routes/updateAppRoute.ts

```diff
@@ -0,0 +1,83 @@
+import {
+    Request,
+    Response,
+} from 'express';
+import Joi from '@hapi/joi';
+import _ from 'lodash/fp';
+
+import db from '../../db';
+import validateRequestFactory from '../../common/services/validateRequest';
+import {
+    prepareAppRouteToRespond,
+} from '../services/prepareAppRoute';
+import {
+    stringifyJSON,
+} from '../../common/services/json';
+import {
+    partialAppRouteSchema,
+} from '../interfaces';
+import {
+    appRouteIdSchema,
+} from '../interfaces';
+
+type UpdateAppRouteRequestParams = {
+    id: string
+};
+
+const validateRequestBeforeUpdateAppRoute = validateRequestFactory([
+    {
+        schema: Joi.object({
+            id: appRouteIdSchema.required(),
+        }),
+        selector: _.get('params'),
+    },
+    {
+        schema: partialAppRouteSchema,
+        selector: _.get('body')
+    },
+]);
+
+const updateAppRoute = async (req: Request<UpdateAppRouteRequestParams>, res: Response) => {
+    await validateRequestBeforeUpdateAppRoute(req, res);
+
+    const {
+        slots: appRouteSlots,
+        ...appRoute
+    } = req.body;
+    const appRouteId = req.params.id;
+
+    const appRoutes = await db.transaction(async (transaction) => {
+        try {
```

Comment:
You don't need it.
Throwing an error directly from the transaction handler function automatically rolls back the transaction, same as returning a rejected promise.

Reply:
Done